content/en/blog/falcosidekick-reponse-engine-part-6-cloud-run.md

```diff
@@ -0,0 +1,189 @@
+
+Let's start by explaining a little bit about what `Cloud Run` is first. `Cloud Run` is a managed compute platform that enables you to run containers that are invocable via requests or events. `Cloud Run` is serverless: it abstracts away all infrastructure management, so you can focus on what matters most — building great applications.
+
+If you want to learn more about them, you can follow this [link](https://cloud.google.com/run/docs).
+
+Today's reference architecture will look like the following picture:
+
+![cloud_run_reference_arch](/docs/images/cloud_run_reference_arch.png)
+
+This demo might be useful for Google Cloud users who might already be using GKE with `Falco` to protect container runtime against malicious behaviors, and wants to take any action for them with `Cloud Run`.
+
+// TOC
+
+## Prerequisites
+
+- gcloud 342.0.0
+- ko 0.8.3
+
+## Tutorial
+
+### Provision GKE (Google Kubernetes Engine) Cluster
+
+We have to create a GKE, so let's start by creating one.
+
+```shell
+$ GKE_CLUSTER_NAME=cloud-run-demo
+$ gcloud container clusters create $GKE_CLUSTER_NAME
+```
+
+### Deploy Cloud Run Function
+
+Once it's finished, we are ready to deploy our Cloud Run function. But before doing that, we have to explain that what the `Cloud Run` function will do. As you can see in the reference architecture image above, this function will delete the pwned Pod. In order to do that, this function should have the appropriate permissions.
+
+There are two approaches to obtain this permissions.
+
+- The first approach would be is creating a Kubernetes `Service Account`, an appropriate `Role` with granted permissions to delete pod resource, and a `RoleBinding` to bind Role to Service Account. Then create the `kubeconfig` file, package it up with the function code while deploying the `Cloud Run` function, and use this file to create a Kubernetes client.
+
+If you want to learn more about the `kubeconfig` files, you can follow this [link](https://ahmet.im/blog/mastering-kubeconfig/).
+
+- The second approach would be is producing a valid `~/.kube/config` with using library called `google.golang.org/api/` within the function code. Because the representation of the valid `~/.kube/config` file is [clientcmd/api/Config](https://pkg.go.dev/k8s.io/client-go@v0.21.1/tools/clientcmd/api#Config) in Go.
+
+We'll go through with the second approach in this demo. Thanks to Scott Blum, his [blog post](https://bionic.fullstory.com/connect-to-google-kubernetes-with-gcp-credentials-and-pure-golang/) is explaining this in more detail. I highly recommend you to check that out.
+
+Let's deploy the function. If you want to take a look at the function code, you can follow this GitHub [repository](github.com:developer-guy/kubernetes-response-engine-based-on-gke-and-cloud-run.git).
```

Comment:
Let's deploy the function. If you want to take a look at the function code, see the [repository](github.com:developer-guy/kubernetes-response-engine-based-on-gke-and-cloud-run.git).

Reply:
thank you so much, done.